Purchases/Networking/Backend.swift

```diff
@@ -32,23 +32,33 @@ class Backend {
     private let httpClient: HTTPClient
     private let subscribersAPI: SubscribersAPI
     private let operationQueue: OperationQueue
+    private let attributionFetcher: AttributionFetcher
 
     private let logInCallbacksCache: CallbackCache<LogInCallback>
     private let offeringsCallbacksCache: CallbackCache<OfferingsCallback>
     private let callbackQueue = DispatchQueue(label: "Backend callbackQueue")
 
     convenience init(apiKey: String,
                      systemInfo: SystemInfo,
-                     eTagManager: ETagManager) {
+                     eTagManager: ETagManager,
+                     attributionFetcher: AttributionFetcher,
+                     dateProvider: DateProvider = DateProvider()) {
```

Comment:
Piping a date provider through so we can test the new SubscriberAttributes being added in this object cc @NachoSoto